### PR TITLE
Feature/db

### PR DESCRIPTION
**Pull request by somaSeb**

### CODE REVIEW

File: pages/index.js [hunk 1]
```diff
@@ -11,18 +11,42 @@ import {
   StyledAllFormButtonsContainer,
 } from "@/styles";
 import FilterCategory from "@/components/FilterCategory/FilterCategory";
-export default function HomePage({
-  transactions,
+import useSWR, {mutate} from "swr";
+
+const fetcher = (url) => fetch(url).then((res) => res.json());
```
Comment (MerleMak) [inline pages/index.js:16]: It makes sense to extract the fetcher to the _app.js to make it globally accessible
https://github.com/neuefische/hh-web-23-5/blob/main/sessions/react-data-fetching/react-data-fetching.md#configuring-swr

File: pages/goals.js [hunk 2]
```diff
@@ -59,14 +64,29 @@ export default function Goals({ transactions, onAddTransaction }) {
...
-  function handleSaveGoal(goalToSave) {
-    if (goalToSave.id) {
-      const originalGoal = goals.find((goal) => goal.id === goalToSave.id);
-      if (originalGoal) {
-        const difference =
-          parseInt(goalToSave.savedAmount) - parseInt(originalGoal.savedAmount);
+
+
+  async function handleSaveGoal(goalToSave) {
+    const isEditing = !!goalToSave._id;
```
Comment (MerleMak) [inline pages/goals.js:70]:
```suggestion
    const isEditing = !!goalToSave;

```

File: pages/goals.js [hunk 3]
```diff
@@ -15,23 +12,31 @@ import {
   StyledCardContainer,
   StyledSavingContainer,
 } from "@/styles";
-export default function Goals({ transactions, onAddTransaction }) {
-  const [goals, setGoals] = useLocalStorageState("goals", {
-    defaultValue: initialGoals,
-  });
+import useSWR, {mutate} from "swr";
+
+const fetcher = (url) => fetch(url).then((res) => res.json());
```
Comment (MerleMak) [inline pages/goals.js:17]: transfer to _app.js

File: pages/dashboard.js [hunk 4]
```diff
@@ -2,12 +2,20 @@ import Navbar from "@/components/Nav/Nav";
...
 import styled from "styled-components";
 import DoughnutComponent from "@/components/Doughnut/Doughnut";
+import useSWR from "swr";
 
 const StyledText = styled.p`
   padding-top: 60px;
 `;
 
-export default function Dashboard({ transactions }) {
+const fetcher = (url) => fetch(url).then((res) => res.json());
```
Comment (MerleMak) [inline pages/dashboard.js:11]: .

File: pages/goals.js (hunk 2 above)
Comment (MerleMak) [reply pages/goals.js:70]: Why not use the Boolean Constructor? its probably a bit more readable?

File: pages/_app.js [hunk 5]
```diff
@@ -1,25 +1,47 @@
 import GlobalStyle from "../styles";
-import { initialTransactions } from "@/ressources/data";
-import { v4 as uuidv4 } from "uuid";
-import useLocalStorageState from "use-local-storage-state";
-import { useState } from "react";
+import useSWR, {mutate} from "swr";
+
+const fetcher = (url) => fetch(url).then((res) => res.json());
```
Comment (MerleMak) [inline pages/_app.js:4]: .

**Review by MerleMak — APPROVED** [review]

**Comment by somaSeb** [comment]
Changes applied. Thanks!

**Pull request merged**